src/Language/GraphQL/Draft/Syntax.hs

```diff
@@ -374,56 +380,57 @@ isNotNull = not . isNullable
 
 -- * Type definition
 
-data TypeDefinition possibleTypes
+data TypeDefinition possibleTypes a
   = TypeDefinitionScalar ScalarTypeDefinition
-  | TypeDefinitionObject ObjectTypeDefinition
-  | TypeDefinitionInterface (InterfaceTypeDefinition possibleTypes)
+  | TypeDefinitionObject (ObjectTypeDefinition a)
+  | TypeDefinitionInterface (InterfaceTypeDefinition possibleTypes a)
   | TypeDefinitionUnion UnionTypeDefinition
   | TypeDefinitionEnum EnumTypeDefinition
-  | TypeDefinitionInputObject InputObjectTypeDefinition
-  deriving (Ord, Show, Eq, Lift, Generic)
-instance Hashable possibleTypes => Hashable (TypeDefinition possibleTypes)
+  | TypeDefinitionInputObject (InputObjectTypeDefinition a)
+  deriving (Ord, Show, Eq, Lift, Generic, Functor)
```

Comment:
I'm not sure I'm super fond of having a `Functor` instance because it's not clear whether it would be better to map over `possibleTypes` or over `a`. How is this used?

Reply:
I had thought of this as well. But since `possibleTypes` is used only in the `InterfaceTypeDefinition` and `a` being used more, I went with having the `Functor` instance for `a`.

Reply:
Okay, fair enough, I have no strong opinion about this.